src/web/vaev-layout/table.cpp

```diff
@@ -1192,15 +1210,19 @@ struct TableFormatingContext : public FormatingContext {
         return {startAt, stopAt};
     }
 
-    Output run(Tree& tree, Box& box, Input input, usize startAtTable, Opt<usize> stopAtTable) {
+    Opt<CacheParametersFromInput> lastInput;
+
+    Output run(Tree& tree, Box& box, Input input, usize startAtTable, Opt<usize> stopAtTable) override {
         // TODO: - vertical and horizontal alignment
         //       - borders collapse
         // TODO: in every row, at least one cell must be an anchor, or else this row is 'skipable'
 
-        // HACK: Quick reset for formating context reuse.
-        //       Proper reset logic to be implemented in a future commit.
-        *this = {};
-        build(tree, box, input);
+        CacheParametersFromInput inputCacheParameters{input};
+        if (not lastInput or lastInput.unwrap() != inputCacheParameters) {
+            // TODO: do we already support different page sizes?
+            lastInput = inputCacheParameters;
+            computeWidthAndHeight(tree, box, lastInput.unwrap());
+        }
```

Comment:
This looks pretty simple and useful for other formatting contexts. Do you think we could generalize this?

Pseudo code:
```c++
auto key = input.cacheKey();
if (fc.lastKey != key) {
    gc.rebuild();
    fc.lastKey = key;
}
```

Reply:
From this we discussed:
Separating the steps of a formatting context into:
- build: non-repeatable for a given FC, does not depend on input
- measure: compute dimensions, depends on input
- run: layout children

and from this, a caching mechanism would always call `measure` again.
However we agreed that fixing `measure` and the caching structure like this could limit the formatting context, so we decided to leave the responsibility of measure and re-measure for the child classes.

Reply:
Ok